File: JobSequencing.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collection;
import java.util.Collections;

class Job{
    char id;
    int deadline;
    int profit;

    public Job(){

    }
    public Job(char id, int deadline, int profit)
    {
        this.id = id;
        this.deadline = deadline;
        this.profit = profit;
    }

    public void printJobScheduling(ArrayList<Job> arr , int n){
        Collections.sort(arr,(a,b) -> (b.profit - a.profit));
        int maxi=0;
        for(int i =0 ; i < n ; i++){
            maxi=Math.max(maxi, arr.get(i).deadline);
        }
        int[] result = new int[maxi+1];
        for(int i =0 ; i <= maxi ;i++){
            result[i]=-1;
        }
        System.out.println();
        int count =0 , maxProfit=0;
        for(int i =0 ; i < n ; i++){
            for(int j = arr.get(i).deadline;j>=1;j--){
                if(result[j]==-1){
                    count+=1;
                    maxProfit+=arr.get(i).profit;
                    result[j]=i;
                    System.out.println(arr.get(i).id);
                    break;

                }
            }
        }

    }
}
class JobSequencing{
    public static void main(String[] args) {
        ArrayList<Job> arr = new ArrayList<Job>();
 
        arr.add(new Job('a', 2, 100));
        arr.add(new Job('b', 1, 19));
        arr.add(new Job('c', 2, 27));
        arr.add(new Job('d', 1, 25));
        arr.add(new Job('e', 3, 15));
       
        // Function call
        System.out.println("Following is maximum "
                           + "profit sequence of jobs");
 
        Job job = new Job();
 
        // Calling function
        job.printJobScheduling(arr, arr.size());
        
    }
}

File: JobSequencing.py
class Job:
    def __init__(self,id,deadline,profit):
        self.id=id
        self.deadline=deadline
        self.profit=profit
            
    
    
    def findJobSequence(jobs):
        jobs.sort(key=lambda x : x.profit,reverse=True)
        maxDeadline=0
        n=len(jobs)
        for i in range(len(jobs)):
            if jobs[i].deadline > maxDeadline:
                maxDeadline=jobs[i].deadline
        resultArr=[-1 for i in range(maxDeadline+1)]
        
        count=0
        maxProfit=0
        
        for i in range(n):
            for j in range(jobs[i].deadline,0,-1):
                if resultArr[j]==-1:
                    count+=1
                    maxProfit+=jobs[i].profit
                    resultArr[j]=jobs[i].id
                    break 
        print("no of jobs that can be executed",count) 
        print("max profit obtained =",maxProfit)    
        print("sequence of jobs is ")
        for i in range(len(resultArr)):
            if resultArr[i]!=-1:
                print(resultArr[i],end="->")   
                    
                
                
        return


if __name__=="__main__" :
    jobs=[]
    jobs.append(Job('a', 2, 100))
    jobs.append(Job('b', 1, 19))
    jobs.append(Job('c', 2, 27))
    jobs.append(Job('d', 1, 25))
    jobs.append(Job('e', 3, 15))
    
    Job.findJobSequence(jobs)
    
           